Userland/Utilities/ls.cpp

```diff
@@ -274,8 +274,8 @@ static size_t print_name(const struct stat& st, DeprecatedString const& name, ch
         printf("%s", end_color);
     }
     if (S_ISLNK(st.st_mode)) {
-        if (path_for_link_resolution) {
-            auto link_destination_or_error = Core::DeprecatedFile::read_link(path_for_link_resolution);
+        if (path_for_link_resolution.has_value()) {
+            auto link_destination_or_error = FileSystem::read_link(path_for_link_resolution.value());
             if (link_destination_or_error.is_error()) {
                 perror("readlink");
```

Comment:
This is subtly broken, instead of calling `perror("readlink")`, now we must `warnln("{}", link_destination_or_error.error())` as the FileSystem API will not set errno but return the Error directly.

Reply:
This makes me want to write a fault-injector, perhaps using ptrace or something, just to see where else things go wrong.